### PR TITLE
Use Swagger to generate rest docs

### PR DESCRIPTION
**Pull request by Xenorith**

### CODE REVIEW

**Comment by Xenorith** [comment]
jenkins, test this please

**Comment by Xenorith** [comment]
jenkins, test this please

**Comment by ZacBlanco** [comment]
jenkins, test this please

**Comment by ZacBlanco** [comment]
jenkins, test this please

**Comment by Xenorith** [comment]
jenkins, test this please

**Comment by Xenorith** [comment]
jenkins, test this please

**Comment by Xenorith** [comment]
oh great it finally passed @ZacBlanco

File: core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientRestServiceHandler.java [hunk 1]
```diff
@@ -410,13 +413,15 @@ public Response scheduleAsyncPersist(@QueryParam("path") final String path) {
...
-      @QueryParam("recursive") final Boolean recursive,
-      @QueryParam("ttxAction") final TtlAction ttlAction) {
+                               @QueryParam("pinned") final Boolean pinned,
+                               @QueryParam("ttl") final Long ttl,
+                               @QueryParam("persisted") final Boolean persisted,
+                               @QueryParam("owner") final String owner,
+                               @QueryParam("group") final String group,
+                               @QueryParam("permission") final Short permission,
+                               @QueryParam("recursive") final Boolean recursive,
+                               @QueryParam("ttxAction") final TtlAction ttlAction) {
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientRestServiceHandler.java]: Could you revert this formatting change?

File: core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientRestServiceHandler.java [hunk 2]
```diff
@@ -395,6 +380,24 @@ public Response scheduleAsyncPersist(@QueryParam("path") final String path) {
...
+    return RestUtils.call(new RestUtils.RestCallable<Void>() {
+      @Override
+      public Void call() throws Exception {
+        Preconditions.checkNotNull(path, "required 'path' parameter is missing");
+        mFileSystemMaster.unmount(new AlluxioURI(path));
+        return null;
+      }
+    });
+  }
+
```
Comment (calvinjia) [inline core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientRestServiceHandler.java]: Could you revert the method ordering change?

File: core/server/proxy/src/main/java/alluxio/proxy/PathsRestServiceHandler.java [hunk 3]
```diff
@@ -122,7 +124,8 @@ public Void call() throws Exception {
    */
   @POST
   @Path(PATH_PARAM + CREATE_FILE)
-  @ReturnType("java.lang.Integer")
+  @ApiOperation(value = "Create a file at the given path, use the id with the streams api",
+                response = java.lang.Integer.class)
```
Comment (calvinjia) [inline core/server/proxy/src/main/java/alluxio/proxy/PathsRestServiceHandler.java]: 4 space indent

**Comment by calvinjia** [comment]
@Xenorith This will only generate docs for the nondeprecated endpoints, but I think that is probably better.

File: core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientRestServiceHandler.java (hunk 1 above)
Comment (Xenorith) [reply core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientRestServiceHandler.java]: this change fixes checkstyle error when building

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
@Xenorith seems like the test failure is similar to the previous one that was flaking. I will merge this since the follow up changes are mostly formatting related.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**